include/bluetooth/mesh/sensor.h

```diff
@@ -64,18 +68,94 @@ enum bt_mesh_sensor_sampling {
 	BT_MESH_SENSOR_SAMPLING_COUNT,
 };
 
+/** Delta threshold type. */
+enum bt_mesh_sensor_delta {
+	/** Value based delta threshold.
+	 *  The delta threshold values are represented as absolute value
+	 *  changes.
+	 */
+	BT_MESH_SENSOR_DELTA_VALUE,
+	/** Percent based delta threshold.
+	 *  The delta threshold values are represented as percentages of their
+	 *  old value (resolution: 0.01 %).
+	 */
+	BT_MESH_SENSOR_DELTA_PERCENT,
+};
+
+/** Sensor sampling cadence */
+enum bt_mesh_sensor_cadence {
+	/** Normal sensor publish interval. */
+	BT_MESH_SENSOR_CADENCE_NORMAL,
+	/** Fast sensor publish interval. */
+	BT_MESH_SENSOR_CADENCE_FAST,
+};
+
+/** Unit for single sensor channel values. */
+struct bt_mesh_sensor_unit {
+	/** English name of the unit, e.g. "Decibel". */
+	const char *name;
+	/** Symbol of the unit, e.g. "dB". */
+	const char *symbol;
+};
+
+struct bt_mesh_sensor_format;
+
+/** Signle sensor channel */
+struct bt_mesh_sensor_channel {
+	/** Format for this sensor channel. */
+	const struct bt_mesh_sensor_format *format;
+#ifdef CONFIG_BT_MESH_SENSOR_LABELS
+	/** Name of this sensor channel. */
+	const char *name;
+#endif
+};
+
+/** Flag indicating this sensor type has a series representation. */
+#define BT_MESH_SENSOR_TYPE_FLAG_SERIES BIT(0)
+
+/** Sensor type. Should only be instantiated in sensor_types.c.
+ *  See sensor_types.h for a list of all defined sensor types.
+ */
+struct bt_mesh_sensor_type {
+	/** Device Property ID. */
+	uint16_t id;
+	/** Flags, @see BT_MESH_SENSOR_TYPE_FLAG_SERIES */
+	uint8_t flags;
+	/** The number of channels supported by this type. */
+	uint8_t channel_count;
+	/** Array of channel descriptors.
+	 *
+	 *  All channels are mandatory and immutable.
+	 */
+	const struct bt_mesh_sensor_channel *channels;
+};
+
+struct bt_mesh_sensor;
+struct bt_mesh_sensor_srv;
+
+#if !defined(CONFIG_BT_MESH_SENSOR_USE_SENSOR_VALUE) || defined(__DOXYGEN__)
+/** Sensor value type representing the value and format of a single channel of
+ *  sensor data.
+ */
+struct bt_mesh_sensor_value {
+	const struct bt_mesh_sensor_format *format;
+	uint8_t raw[CONFIG_BT_MESH_SENSOR_CHANNEL_ENCODED_SIZE_MAX];
```

Comment:
@omkar3141 has pointed out that in the future, variably sized sensor channels might be supported. My design of statically allocating ENCODED_SIZE_MAX bytes in every struct will not work very well if ENCODED_SIZE_MAX starts to become a very high number like 32 or 64 or even bigger. This will cause a lot of wasted memory. Should I change this to a pointer instead for future-proofing, or should that wait until we actually know that we need it?

Reply:
How the memory for the value will be declared then? Like for net buf?
https://github.com/zephyrproject-rtos/zephyr/blob/eeb6bf7dd998c8c191f58df692097bfcfaf1e3bf/include/zephyr/net/buf.h#L42-L49

Reply:
Yes that was my idea

Reply:
Probably ok, but declaring `bt_mesh_sensor_value` for multichannel sensors will be more complicated.

Reply:
I realize now that this wouldn't solve my problem either. Since the size of the allocated `raw` buffer will have to be known at compile time, we either way need to allocate MAX for every incoming message. Even with `NET_BUF`-like macros. And, like you say, it would complicate things for multichannel sensors. I'll keep it as-is for now I think.

Reply:
ENCODED_SIZE_MAX is okay, because you would need it anyway.